### ISSUE TITLE
Service msg don't have action...

### ISSUE THREAD

**Issue by rockneurotiko** [yagop/telegram-bot#201]
When I wrote the service message processing, I was using the `test` branch of tg and tgl (that don't exist now) and message was much more better, with the messages thet tg gives now, we can't know what service is :S 

In test was:
```
{
  action = {
    type = "chat_add_user",
    user = {
      access_hash = 1,
      first_name = "",
      flags = 257,
      id = 0,
      phone = "0",
      print_name = "",
      type = "user",
      username = ""
    }
  },
  date = 1432798959,
  flags = 8451,
  from = {
    first_name = "",
    flags = 264,
    id = 0,
    last_name = "",
    phone = "",
    print_name = "",
    type = "user",
    username = ""
  },
  id = "268317",
  out = true,
  service = true,
  to = {
    flags = 256,
    id = 13167526,
    members_num = 4,
    print_name = "oaoeaoe",
    title = "oaoeaoe",
    type = "chat"
  },
  unread = true
}
```

Now is:
```
{
  date = 1432798772,
  flags = 16,
  from = {
    first_name = "",
    flags = 144,
    id = 0,
    last_name = "",
    phone = "0",
    print_name = "",
    type = "user"
  },
  id = "268306",
  out = true,
  service = true,
  to = {
    flags = 16,
    id = 13167526,
    members = {
      {
        access_hash = 1,
        first_name = "",
        flags = 16,
        id = 0,
        last_name = "",
        print_name = "",
        type = "user"
      },
      {
        access_hash = 1,
        first_name = "",
        flags = 16,
        id = 0,
        last_name = "",
        print_name = "",
        type = "user"
      },
      [0] = {
        access_hash = 1,
        first_name = "Bot",
        flags = 528,
        id = 0,
        phone = "0",
        print_name = "Bot",
        type = "user"
      }
    },
    members_num = 3,
    print_name = "oaoeaoe",
    title = "oaoeaoe",
    type = "chat"
  },
  unread = true
}
```

They don't give any action information, even don't give the username.. I don't know why, it worked well in test branch, but they removed that branch but don't take that improvements... XD
Right now I think that it's almost imposible to know the service type :S 
@yagop some idea?

**Comment by yagop** [comment]
I was testing it yesterday and had the same problem. I was going to ask you about it.

:cry:

**Comment by rockneurotiko** [comment]
Yeah, Daus told me this morning... and I was like.. But it's working for me!!!
Yeah... in the test version... That don't exist anymore XD

I will ask in tg/tgl, because it was a really good info :S

**Comment by yagop** [comment]
Tested on latest tg version and its working! :smiley: 

```
{
  action = {
    type = "chat_add_user",
    user = {
      access_hash = 1,
      first_name = "Bot",
      flags = 257,
      id = 123456,
      phone = "123456",
      print_name = "Bot",
      type = "user",
      username = "username"
    }
  },
[...]
```

**Comment by rockneurotiko** [comment]
They read us! jajajajajajajaaj
So, is working the service when add a new user (I can't test it in work :P)? ^^

**Comment by soend** [comment]
Looks like there is no username field. Getting this error with the latest tg:
*** lua: plugins/service_entergroup.lua:32: attempt to concatenate field 'username' (a nil value)

Could use print_name there instead maybe?
`local to_username = '@' .. msg.action.user.print_name`

**Comment by rockneurotiko** [comment]
@soend I'll write to let it empty if it's not setted, because in the template you specify the username, we can't pur the username, but always can extend the template to accept the print_name :)

**Comment by soend** [comment]
Ok, that sounds like a solution also. Keep up the good work :D

**Comment by rockneurotiko** [comment]
@soend fixed in ea9664aad6ae7 and I close this because tg supports action again :smile:
